src/components/Table/DataTable/compenents/el-data-table/el-data-table.vue

```diff
@@ -1106,6 +1106,12 @@ export default {
       this.page = val
       this.getList()
     },
+    handleSelectAll(e, canSelect) {
+      this.loading = true
+      this.selectStrategy.onSelectAll(e, canSelect, () => {
+        this.loading = false
+      })
+    },
     /**
      * 切换某一行的选中状态，如果使用了第二个参数，则是设置这一行选中与否
      *
```

Comment:
There seem to be some inconsistencies between the two versions of your code. Please review them carefully. I don't have time constraints currently. Could you please send the full details?